hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDTableServiceClient.java

```diff
@@ -86,13 +93,29 @@ protected HoodieWriteMetadata<JavaRDD<WriteStatus>> compact(String compactionIns
   @Override
   protected HoodieWriteMetadata<JavaRDD<WriteStatus>> logCompact(String logCompactionInstantTime, boolean shouldComplete) {
     HoodieSparkTable<T> table = HoodieSparkTable.create(config, context);
+
+    // Check if a commit or compaction instant with a greater timestamp is on the timeline.
+    // If a instant is found then abort log compaction, since it is no longer needed.
```

Comment:
can you help me understand why do we need to add this? 
why can't we trigger log compaction if there are higher commits ?